### PR TITLE
Add reviews feature tests

### PR DESCRIPTION
**Pull request by kasawyer**
Added feature tests for editing and deleting reviews, updated review functionality and podcast show page.

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -16,23 +16,26 @@ def create
 
   def edit
     @podcast = Podcast.find(params[:podcast_id])
-    @review = Review.find(params[:id])
-    @new_review = @review
+    @edit_review = Review.find(params[:id])
+    @new_review = @edit_review
```
Comment (lilybarrett) [inline app/controllers/reviews_controller.rb:20]: Couldn't you just do `@new_review = Review.find(params[:id])`?

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -16,23 +16,26 @@ def create
...
+    if @edit_review.update(review_params)
       flash[:notice] = "Review updated successfully"
+      @new_review = Review.new(review_params)
+      @new_review.podcast = @podcast
+      @new_review.user = current_user
       redirect_to podcast_path(@podcast)
     else
-      flash.now[:notice] = "Review not updated"
+      flash.now[:notice] = @edit_review.errors.full_messages.to_sentence
+      @podcast = Podcast.find(params[:podcast_id])
```
Comment (lilybarrett) [inline app/controllers/reviews_controller.rb:36]: Your `update` path, remember, doesn't actually _show_ anything. It looks like some of this code (for the ordered `@reviews`, for instance) belongs in your `show` action, which is the view where your reviews are actually displayed. Please consider & try to clean this up in a future PR!

File: app/models/review.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@ class Review < ApplicationRecord
   belongs_to :user
   belongs_to :podcast
 
-  validates :rating, presence: true, numericality: { in: 1..5 }
+  validates :rating, presence: true, numericality: true, inclusion: {
+    in: 1..5, message: "must be between 1 - 5"
+  }
```
Comment (lilybarrett) [inline app/models/review.rb:7]: Good!

File: spec/features/08_user_edits_review_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+feature 'visitors can edit reviews' do
+  let! (:npr) { Provider.create(name: "NPR") }
+  let! (:fresh_air) { Podcast.create(name: "Fresh Air", provider: npr) }
+  let! (:user) do
+    User.create(
```
Comment (lilybarrett) [inline spec/features/08_user_edits_review_spec.rb:7]: I would really like to see you using Factory Girl, in place of creating the objects "by hand," in a future PR.

File: app/controllers/reviews_controller.rb (hunk 1 above)
Comment (kasawyer) [reply app/controllers/reviews_controller.rb:20]: We actually don't even need line 20. I can delete it.

File: app/controllers/reviews_controller.rb (hunk 2 above)
Comment (kasawyer) [reply app/controllers/reviews_controller.rb:36]: Will do! Thanks!

File: spec/features/08_user_edits_review_spec.rb (hunk 4 above)
Comment (kasawyer) [reply spec/features/08_user_edits_review_spec.rb:7]: Ok, we can do that. Thanks!

**Pull request merged**